### DISCUSSION TITLE
How to build Firefly

### DISCUSSION THREAD

**Discussion by matejdro** [firefly-iii/firefly-iii#8933] [Firefly III] [answered]
Hello,

it seems that all deployment guides now only give us the option to download official release from Github.

I cannot find any guide on how to properly build and deploy Firefly from source. Is this documented anywhere?

**Comment by JC5** [comment]
Not yet! For the time being, you can use the release action as a workflow. I will write something for it.

https://github.com/firefly-iii/firefly-iii/blob/main/.github/workflows/release.yml

**Comment by JC5** [comment, marked as answer]
Short version.

```
composer install --no-dev --no-scripts --no-plugins
npm run prod  --workspace=v1
npm run build --workspace=v2
```

Then run the installation instructions from the documentation.

**Reply by matejdro** [reply]
Thank you, it worked!
